xrpl/utils/tx_parser/utils.py

```diff
@@ -0,0 +1,847 @@
+        """Create a new class with variable name.
+
+        Args:
+            name (str): field name
+
+        Returns:
+            Any: A class with the name of a field. E.g. 'OwnerCount', 'Balance', ….
+        """
+
+        class NewClass:
+            pass
+
+        NewClass.__name__ = name
+
+        return NewClass
+
+    field_state_map: Dict[str, Union[NewFields, FinalFields, PreviousFields]] = {
+        "NewFields": NewFields(),
+        "FinalFields": FinalFields(),
+        "PreviousFields": PreviousFields(),
+    }
+
+    field = field_state_map[field_state]
+
+    for field_name, field_value in fields.items():
+        # if 'field_value' is a dictionary
+        if isinstance(field_value, dict):
+            # if 'field_value' is a issued currency amount
+            if field_name in ["Balance", "LowLimit", "HighLimit"]:
+                field_value = Balance(
+                    counterparty=field_value["issuer"],
+                    currency=field_value["currency"],
+                    value=field_value["value"],
+                )
+            else:  # if 'field_value' type is dict but is no issued currency amount.
+                factured_field_name = factory(field_name)
+                for field_value_key, field_value_value in field_value.items():
+                    setattr(factured_field_name, field_value_key, field_value_value)
+                field_value = factured_field_name
+        # set attributes to field state
+        setattr(field, field_name, field_value)
+
+    return field
+
+
+def normalize_node(affectedNode: Dict[str, Any]) -> NormalizedNode:
+    """Affected node to a standard format.
+
+    Args:
+        affectedNode (Dict[str, Any]):
+            Affected node.
+
+    Returns:
+        NormalizedNode:
+            NormalizedNode object.
+    """
+    diff_type = list(affectedNode)[0]
+    node = affectedNode[diff_type]
+
+    normalized_node = NormalizedNode(
+        diffType=diff_type,
+        entryType=node["LedgerEntryType"],
+        ledgerIndex=node["LedgerIndex"],
+        newFields=field_factory("NewFields", node["NewFields"])
+        if ("NewFields" in node)
+        else None,
+        finalFields=field_factory("FinalFields", node["FinalFields"])
+        if ("FinalFields" in node)
+        else None,
+        previousFields=field_factory("PreviousFields", node["PreviousFields"])
+        if ("PreviousFields" in node)
+        else None,
+    )
+
+    return normalized_node
+
+
+def normalize_nodes(metadata: Dict[str, Any]) -> Iterable[NormalizedNode]:
+    """Normalize nodes.
+
+    Args:
+        metadata (Dict[str, Any]): Transctions metadata
+
+    Returns:
+        Iterable[NormalizedNode]: The normalized nodes
+    """
+    affected_nodes = metadata["meta"]["AffectedNodes"]
+    if not affected_nodes:
+        return []
+
+    return map(normalize_node, affected_nodes)
+
+
+"""
+#####################
+Utils balance_changes
+#####################
+"""
+
+
+class Balance:
```

Comment:
This might be the same as an `IssuedCurrencyAmount`. 

You may also want to see if `IssuedCurrency.py` is a useful file for you.

Reply:
Or maybe not because the `counterparty` doesn't correspond to the currency, it corresponds to the payment?